typescript/sdk/src/warp/WarpCore.ts

```diff
@@ -44,6 +48,13 @@ export interface WarpCoreOptions {
   routeBlacklist?: RouteBlacklist;
 }
 
+function isXERC20Adapter<T>(x: ITokenAdapter<T>): x is IHypXERC20Adapter<T> {
+  return (
+    !!(x as any as IHypXERC20Adapter<T>).belowBurnLimit &&
+    !!(x as any as IHypXERC20Adapter<T>).belowMintLimit
+  );
+}
```

Comment:
Rather than trying to introspect this way, branch based on the TokenStandard (i.e. if the standard is `XERC20` or `XERC20Lockbox`